ykutil/src/addr.rs

```diff
@@ -5,16 +5,66 @@ use libc::{self, c_void, Dl_info};
 use phdrs::objects;
 use std::mem::MaybeUninit;
 use std::{
-    convert::TryFrom,
+    convert::{From, TryFrom},
     ffi::CStr,
     path::{Path, PathBuf},
-    ptr::null,
 };
 
-pub fn dladdr(vaddr: usize) -> Result<Dl_info, ()> {
+/// A Rust wrapper around `libc::Dl_info` using FFI types.
+///
+/// The strings inside are handed out by the loader and can (for now, since we don't support
+/// dlclose) be considered of static lifetime.
+#[derive(Debug)]
+pub struct DLInfo {
+    dli_fname: Option<&'static CStr>,
```

Comment:
I *think* `&'static` is probably OK here but it does require us to guarantee that anything we reference can never be unloaded. Is that true? Or, at least, should we make that requirement explicit to the user somewhere?

Reply:
It's discussed in the docstring for this struct. dlopen/dlclose would cause us some problems.

There are many places in the JIT that assume dlopen/dlclose don't happen right now. Later I think we can wrap the libc versions of these functions and call a cache invalidation routines. But I don't want to go there at this moment in time. Is that OK?

Reply:
OK.